### PR TITLE
Rename MeterSharedState with MeterProviderSharedState

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

File: sdk/src/main/java/io/opentelemetry/sdk/trace/TracerSdk.java [hunk 1]
```diff
@@ -32,11 +32,13 @@
 public final class TracerSdk implements Tracer {
   private static final BinaryFormat<SpanContext> BINARY_FORMAT = new BinaryTraceContext();
   private static final HttpTextFormat<SpanContext> HTTP_TEXT_FORMAT = new HttpTraceContext();
-  private final TracerSharedState sharedState;
+  private final TracerSharedState meterProviderSharedState;
```
Comment (jkwatson) [inline sdk/src/main/java/io/opentelemetry/sdk/trace/TracerSdk.java]: this doesn't seem right. Did you over-reach on the renaming of the variables?

Comment (bogdandrutu) [reply sdk/src/main/java/io/opentelemetry/sdk/trace/TracerSdk.java]: fixed, sorry my `sed` skills failed.

**Comment by jkwatson** [comment]
This seems fine, but I'm not sure why it's necessary. Is MeterSharedState not clear as to what it's about?

**Comment by bogdandrutu** [comment]
@jkwatson I will also have a MeterSharedState which is the meter shared state between instruments which includes library info and other meter specific things.

**Comment by jkwatson** [comment]
looks like it needs a rebase

**Pull request merged**